mllam_data_prep/create_dataset.py

```diff
@@ -120,28 +121,56 @@ def create_dataset(config: Config):
 
     output_config = config.output
     output_coord_ranges = output_config.coord_ranges
+    chunking_config = config.output.chunking or {}
 
     dataarrays_by_target = defaultdict(list)
 
     for dataset_name, input_config in config.inputs.items():
         path = input_config.path
-        variables = input_config.variables
+        variables = input_config.variables or None
+        derived_variables = input_config.derived_variables or None
         target_output_var = input_config.target_output_variable
         expected_input_attributes = input_config.attributes or {}
         expected_input_var_dims = input_config.dims
 
         output_dims = output_config.variables[target_output_var]
 
-        logger.info(f"Loading dataset {dataset_name} from {path}")
-        try:
-            ds = load_and_subset_dataset(fp=path, variables=variables)
-        except Exception as ex:
-            raise Exception(f"Error loading dataset {dataset_name} from {path}") from ex
-        _check_dataset_attributes(
-            ds=ds,
-            expected_attributes=expected_input_attributes,
-            dataset_name=dataset_name,
-        )
+        if variables:
+            logger.info(f"Loading dataset {dataset_name} from {path} and subsetting")
+            try:
+                ds = load_and_subset_dataset(
+                    fp=path, variables=variables, chunking=chunking_config
+                )
+            except Exception as ex:
```

Comment:
I have now removed the exception wrapping for `subset_dataset` and `derive_variables`. But I have kept it wrapped around the `load_input_dataset` function call since I didn't want to have to pass both `dataset_name` and `path` to the `load_input_dataset` function. Let me know what you think.

Reply:
Looks good to me:)